File: kotlin/rna-transcription/src/main/kotlin/RnaTranscription.kt
fun transcribeToRna(dna: String): String = dna.map {
        when(it) {
            'G' -> 'C'
            'C' -> 'G'
            'T' -> 'A'
            'A' -> 'U'
            else -> throw IllegalArgumentException("Invalid nucleotide: $it")
        }
    }.joinToString("")

File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(defn to-rna [dna]
  (let [rna-nucleotides (map '{\G \C, \C \G, \T \A, \A \U} dna)]
    (if (not-any? nil? rna-nucleotides)
      (clojure.string/join "" rna-nucleotides)
      (throw (AssertionError. "invalid DNA nucleotide")))))
